dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java

```diff
@@ -93,19 +122,40 @@ public void synthesizeAndDownloadFeatures(List<String> mavenCoords, String dLoca
             fine("The remote repository is missing the following artifacts: " + missingFeatureList.toString());
             throw ExceptionUtils.createByKey("ERROR_FAILED_TO_DOWNLOAD_ASSETS_FROM_REPO", "required", "feature(s)", repo);
         } else {
+            List<String> result = new ArrayList<String>();
+
+            final List<Future<?>> futures = new ArrayList<>();
             // we have downloaded mavenCoords.length amount of features.
             double individualSize = progressBar.getMethodIncrement("downloadArtifacts") / mavenCoords.size();
             progressBar.updateMethodMap("downloadArtifact", individualSize);
             logger.info(Messages.INSTALL_KERNEL_MESSAGES.getMessage("MSG_BEGINNING_DOWNLOAD_FEATURES"));
             for (String coords : mavenCoords) {
-                synthesizeAndDownload(coords, "esa", dLocation, repository, false);
-                synthesizeAndDownload(coords, "pom", dLocation, repository, false);
+                Future<?> future1 = esaDownload(coords, dLocation, repository);
+                futures.add(future1);
+                Future<?> future2 = pomDownload(coords, dLocation, repository);
+                futures.add(future2);
+//                synthesizeAndDownload(coords, "esa", dLocation, repository, false);
+//                synthesizeAndDownload(coords, "pom", dLocation, repository, false);
+            }
 
-                // update progress bar, drain the downloadArtifacts total size
-                updateProgress(individualSize);
-                progressBar.updateMethodMap("downloadArtifacts", progressBar.getMethodIncrement("downloadArtifacts") - individualSize);
-                progressBar.manuallyUpdate();
-                fine("Finished downloading artifact: " + coords);
+            for (int i = 0; i < futures.size(); i++) {
```

Comment:
Unless we need to serially wait for each download i suggest going through the entire list before sleeping.
Something like:

```
while (!futures.isEmpty())
      {
         Iterator<Future<String>> iter = futures.iterator();

         while (iter.hasNext())
         {
            boolean updateProgress = false;
            Future<String> future = iter.next();
            if (future.isDone())
            {
               String downloadedCoords = future.get();
               if (updateProgress)
               {
                  // update progress bar, drain the downloadArtifacts total size
                  updateProgress(individualSize);
                  progressBar.updateMethodMap("downloadArtifacts",
                        progressBar.getMethodIncrement("downloadArtifacts") - individualSize);
                  progressBar.manuallyUpdate();
                  fine("Finished downloading artifact: " + downloadedCoords);
               }
               updateProgress = !updateProgress;
               iter.remove();
            }
         }

         logger.info("Downloading " + futures.size() + " artifacts");
         Thread.sleep(300);
      }
```

This way time will be spent checking the other futures in the list instead of sleeping.